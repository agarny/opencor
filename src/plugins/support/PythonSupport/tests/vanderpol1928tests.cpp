/*******************************************************************************

Copyright (C) The University of Auckland

OpenCOR is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

OpenCOR is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program. If not, see <https://gnu.org/licenses>.

*******************************************************************************/

//==============================================================================
// Python support van der Pol 1928 tests
//==============================================================================

#include "../../../../tests/src/testsutils.h"

//==============================================================================

#include "vanderpol1928tests.h"

//==============================================================================

#include <QtTest/QtTest>

//==============================================================================

void VanDerPol1928Tests::tests()
{
    // Some tests to make sure that the van der Pol 1928 model works fine

    QStringList output;

    QVERIFY(!OpenCOR::runCli({ "-c", "PythonShell", OpenCOR::fileName("src/plugins/support/PythonSupport/tests/data/vanderpol1928tests.py") }, output));
    QCOMPARE(output, OpenCOR::fileContents(OpenCOR::fileName("src/plugins/support/PythonSupport/tests/data/vanderpol1928tests.out")));
}

//==============================================================================

QTEST_APPLESS_MAIN(VanDerPol1928Tests)

//==============================================================================
// End of file
//==============================================================================
