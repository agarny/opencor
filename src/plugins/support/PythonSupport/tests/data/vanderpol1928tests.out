---------------------------------------------------------------------
                       van der Pol 1928 model
---------------------------------------------------------------------
 - CVODE:
    - Constants:
       - main/epsilon = [ 1.0, 1.0, 1.0, ..., 1.0, 1.0, 1.0 ]
    - States:
       - main/x = [ -2.0, -1.5, -0.3, ..., 0.6, -1.6, -1.8 ]
       - main/y = [ 0.0, 0.8, 1.8, ..., -1.5, -1.8, 0.5 ]
    - Rates:
       - main/x/prime = [ 0.0, 0.8, 1.8, ..., -1.5, -1.8, 0.5 ]
       - main/y/prime = [ 2.0, 0.5, 2.0, ..., -1.6, 4.6, 0.5 ]
    - Algebraic: empty
 - Euler (forward):
    - Constants:
       - main/epsilon = [ 1.0, 1.0, 1.0, ..., 1.0, 1.0, 1.0 ]
    - States:
       - main/x = [ -2.0, -1.5, -0.3, ..., 1.4, 0.1, -2.0 ]
       - main/y = [ 0.0, 0.8, 1.8, ..., -0.8, -2.0, -0.6 ]
    - Rates:
       - main/x/prime = [ 0.0, 0.8, 1.8, ..., -0.8, -2.0, -0.6 ]
       - main/y/prime = [ 2.0, 0.5, 2.0, ..., -0.6, -2.1, 3.7 ]
    - Algebraic: empty
 - Heun:
    - Constants:
       - main/epsilon = [ 1.0, 1.0, 1.0, ..., 1.0, 1.0, 1.0 ]
    - States:
       - main/x = [ -2.0, -1.5, -0.3, ..., 0.6, -1.7, -1.8 ]
       - main/y = [ 0.0, 0.8, 1.8, ..., -1.6, -1.7, 0.6 ]
    - Rates:
       - main/x/prime = [ 0.0, 0.8, 1.8, ..., -1.6, -1.7, 0.6 ]
       - main/y/prime = [ 2.0, 0.5, 2.0, ..., -1.7, 4.7, 0.5 ]
    - Algebraic: empty
 - Runge-Kutta (2nd order):
    - Constants:
       - main/epsilon = [ 1.0, 1.0, 1.0, ..., 1.0, 1.0, 1.0 ]
    - States:
       - main/x = [ -2.0, -1.5, -0.3, ..., 0.6, -1.7, -1.8 ]
       - main/y = [ 0.0, 0.8, 1.8, ..., -1.6, -1.7, 0.6 ]
    - Rates:
       - main/x/prime = [ 0.0, 0.8, 1.8, ..., -1.6, -1.7, 0.6 ]
       - main/y/prime = [ 2.0, 0.5, 2.0, ..., -1.7, 4.7, 0.5 ]
    - Algebraic: empty
 - Runge-Kutta (4th order):
    - Constants:
       - main/epsilon = [ 1.0, 1.0, 1.0, ..., 1.0, 1.0, 1.0 ]
    - States:
       - main/x = [ -2.0, -1.5, -0.3, ..., -1.6, -0.4, 1.8 ]
       - main/y = [ 0.0, 0.8, 1.8, ..., 0.8, 1.7, 1.3 ]
    - Rates:
       - main/x/prime = [ 0.0, 0.8, 1.8, ..., 0.8, 1.7, 1.3 ]
       - main/y/prime = [ 2.0, 0.5, 2.0, ..., 0.5, 1.8, -4.6 ]
    - Algebraic: empty
